src/mixins/amalgamation-mixin.ts

```diff
@@ -315,6 +326,79 @@ export default class AmalgamationMixin extends Vue {
+    return this.getAmalgamatingBusinesses.find(business =>
+      (business.role === AmlRoles.HOLDING || business.role === AmlRoles.PRIMARY)
+    )
+  }
+
+  /**
+   * (Re)fetches the holding/primary business' data and updates the prepopulated data.
+   * @param business The holding/primary business.
+   * @param isNew Whether the business is new (ie, set by user, not restored from draft).
+   */
+  async updatePrepopulatedData (business: AmalgamatingBusinessIF, isNew = false): Promise<void> {
+    // safety check
+    if (!business || business.type !== AmlTypes.LEAR) {
+      throw new Error('updatePrepopulatedData(): invalid business')
+    }
+
+    // first, fetch new directors/share structure/auth info
+    // NB - business addresses have already been fetched
+    // NB - make all API calls concurrently without rejection
+    // NB - if any call failed, that item will be null
+    const [ directors, shareStructure, authInfo ] =
+      await Promise.allSettled([
+        LegalServices.fetchDirectors(business.identifier),
+        LegalServices.fetchShareStructure(business.identifier),
+        AuthServices.fetchAuthInfo(business.identifier)
+      ]).then(results => results.map((result: any) => result.value || null))
+
+    // check for errors before changing anything
+    if (!directors) throw new Error('Unable to fetch directors')
+    if (!shareStructure) throw new Error('Unable to fetch share structure')
+    if (!authInfo) throw new Error('Unable to fetch auth info')
+
+    // unset previous holding/primary business, if any
+    const previous = this.getAmalgamatingBusinesses.find((b: AmalgamatingBusinessIF) =>
+      b.role === AmlRoles.HOLDING || b.role === AmlRoles.PRIMARY
+    )
+    if (previous) {
+      previous.role = AmlRoles.AMALGAMATING
+    }
+
+    // set this business as the new holding/primary business
+    if (this.isAmalgamationFilingVertical) business.role = AmlRoles.HOLDING
+    if (this.isAmalgamationFilingHorizontal) business.role = AmlRoles.PRIMARY
+
+    // overwrite office addresses
+    this.setOfficeAddresses(business.addresses)
+
+    // overwrite directors (but keep completing party, if there is one)
+    const completingParty = this.getAddPeopleAndRoleStep.orgPeople.find((p: OrgPersonIF) =>
+      p.roles.some(role => role.roleType === RoleTypes.COMPLETING_PARTY)
+    )
+    if (completingParty) this.setOrgPersonList([ ...directors, completingParty ])
+    else this.setOrgPersonList(directors)
+
+    // overwrite share structure
+    this.setShareClasses(shareStructure.shareClasses)
+
+    // overwrite business contact -- only when user has marked new holding/primary business,
+    // otherwise leave existing data from restored draft
+    if (isNew) {
+      this.setBusinessContact({
+        email: authInfo.contacts[0].email,
+        confirmEmail: authInfo.contacts[0].email,
+        phone: authInfo.contacts[0].phone,
+        extension: authInfo.contacts[0].extension
+      })
+    }
```

Comment:
In other words, if we're loading this info from the draft, don't overwrite the business contact. (Leave what the user previously saved.)

But if the user just set a new holding/primary business, overwrite the business contact from the new business' data.